rasa/nlu/featurizers/sparse_featurizer/regex_featurizer.py

```diff
@@ -160,23 +291,60 @@ def load(
         model_dir: Optional[Text] = None,
         model_metadata: Optional[Metadata] = None,
         cached_component: Optional["RegexFeaturizer"] = None,
+        should_finetune: bool = False,
         **kwargs: Any,
     ) -> "RegexFeaturizer":
-
+        """Loads a previously trained component.
+
+        Args:
+            meta: Configuration of trained component.
+            model_dir: Path where trained pipeline is stored.
+            model_metadata: Metadata for the trained pipeline.
+            cached_component: Previously cached component(if any).
+            should_finetune: Indicates whether to load the component for further finetuning.
+            **kwargs:
+        """
         file_name = meta.get("file")
-        regex_file = os.path.join(model_dir, file_name)
 
-        if os.path.exists(regex_file):
-            known_patterns = rasa.shared.utils.io.read_json_file(regex_file)
-            return RegexFeaturizer(meta, known_patterns=known_patterns)
-        else:
-            return RegexFeaturizer(meta)
+        patterns_file_name = Path(model_dir).joinpath(file_name + ".patterns.pkl")
```

Comment:
```suggestion
        patterns_file_name = Path(model_dir) / (file_name + ".patterns.pkl")
```

Reply:
same for the others

Reply:
Tbh I like the non-ambiguous and descriptive nature of `joinpath`. Just a personal preference I think 😅  . Changing it to be consistent with other parts of code(in case they are also using the same syntax 😉  )

Reply:
> Just a personal preference I think

Absolutely true . Should've marked it as optional!